src/vfs/public.c

```diff
@@ -152,6 +161,36 @@ static bool vfs_mkparents_recurse(const char *path) {
 
 	return vfs_mkparents_recurse(p) && vfs_mkdir(p);
 }
+*/
+static bool vfs_mkparents_recurse(const char *path) {
+
+    char p[strlen(path) + 1];
+    VFSInfo i;
+    strcpy(p,path);
+    while(*p){
+        char *psep = strrchr(p, VFS_PATH_SEPARATOR);
+
+        if(!psep){
+            //current parent is root
+            return true;
+        }
+
+        *psep = 0;
+
+        i = vfs_query(p);
+
+        if(i.error){
+            return false;
+        }else if(i.exists){
+            return true;
+        }else if(!vfs_mkdir(p)){
+            return true;
```

Comment:
The logic here seems wrong as well. `vfs_mkdir` returns `true` on success, in which case we're done. If it fails, though, that's when we have to keep going and repeat the process for the parents — **and then retry `mkdir`ing the original path**. That part in bold is likely the main reason why the original function is recursive — the algorithm requires a stack. To remove the recursion I'd think you have to iterate forwards instead, but whether that is a win is not clear. Such an algorithm would have the worst-case performance in the "no-op" case (when all parents already exist).